plugins/pulp_docker/plugins/importers/v1_sync.py

```diff
@@ -218,88 +203,61 @@ def find_and_read_ancestry_file(image_id, parent_dir):
-            # its metadata
-            size = metadata.get('Size')
-            # an older version of docker used a lowercase "p"
-            parent = metadata.get('parent', metadata.get('Parent'))
-            model = models.Image(image_id, parent, size)
-            unit = self.get_conduit().init_unit(model.TYPE_ID, model.unit_key, model.unit_metadata,
-                                                model.relative_path)
-
-            self.move_files(unit)
-            _logger.debug('saving image %s' % image_id)
-            self.get_conduit().save_unit(unit)
-
-        _logger.debug('updating tags for repo %s' % self.get_repo().id)
-        tags.update_tags(self.get_repo().id, self.parent.tags)
-
-    def move_files(self, unit):
+        with open(os.path.join(self.get_working_dir(), item.image_id, 'json')) as json_file:
+            metadata = json.load(json_file)
+        # at least one old docker image did not have a size specified in
+        # its metadata
+        size = metadata.get('Size')
+        # an older version of docker used a lowercase "p"
+        parent = metadata.get('parent', metadata.get('Parent'))
+        item.parent_id = parent
+        item.size = size
+        item.set_content(os.path.join(self.get_working_dir(), item.image_id))
+
+        item.save()
+        repo_controller.associate_single_unit(self.get_repo(), item)
+
+    def finalize(self):
         """
-        For the given unit, move all of its associated files from the working
-        directory to their permanent location.
-
-        :param unit:    a pulp unit
-        :type  unit:    pulp.plugins.model.Unit
+        Update the tags on the repository object. This method is called after process_main has been
+        called. This will be called even if process_main or initialize raises an exceptions.
         """
-        image_id = unit.unit_key['image_id']
-        _logger.debug('moving files in to place for image %s' % image_id)
-        source_dir = os.path.join(self.working_dir, image_id)
-        try:
-            os.makedirs(unit.storage_path, mode=0755)
-        except OSError, e:
-            # it's ok if the directory exists
-            if e.errno != errno.EEXIST:
-                _logger.error('could not make directory %s' % unit.storage_path)
-                raise
-
-        for name in ('json', 'ancestry', 'layer'):
-            shutil.move(os.path.join(source_dir, name), os.path.join(unit.storage_path, name))
+        super(SaveImages, self).finalize()
+        # Get an updated copy of the repo so that we can update the tags
+        repo = self.get_repo()
+        _logger.debug('updating tags for repo {repo_id}'.format(repo_id=repo.repo_id))
+        if self.parent.tags:
+            new_tags = tags.generate_updated_tags(repo.scratchpad, self.parent.tags)
+            platform_models.Repository.objects(repo_id=repo.repo_id).\
+                update_one(set__scratchpad__tags=new_tags)
```

Comment:
I love how easy this stuff is now! This would have been half punctuation with pymongo.